spec/models/developer_spec.rb

```diff
@@ -37,11 +37,24 @@
     end
   end
 
-  it 'should validate username uniqueness' do
-    FactoryGirl.create(:developer, username: developer.username)
-    dup_developer = FactoryGirl.build(:developer, username: developer.username)
-    expect(dup_developer).to_not be_valid
-    expect(dup_developer.errors.messages[:username]).to eq ['has already been taken']
+  context 'validates usernames' do 
+    it 'should validate username uniqueness' do
+      FactoryGirl.create(:developer, username: developer.username)
+      dup_developer = FactoryGirl.build(:developer, username: developer.username)
+      expect(dup_developer).to_not be_valid
+      expect(dup_developer.errors.messages[:username]).to eq ['has already been taken']
+    end
+
+    it 'should allow alphanumeric usernames' do
+      alpha_numeric_developer = FactoryGirl.build(:developer, username: "johnsmith123")
+      expect(alpha_numeric_developer).to be_valid
+    end
+
+    it 'and deny usernames that contain other characters' do
```

Comment:
Can we change `and` to `should` here? This project has a checked-in `.rspec` that runs specs in random order, so the meaning of  `and` as a conjunction will be lost much of the time.